securebanking-openbanking-uk-rcs-api/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/converters/DomesticVrpPaymentConsentDetailsConverter.java

```diff
@@ -0,0 +1,219 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.securebanking.openbanking.uk.rcs.converters;
+
+import com.forgerock.securebanking.common.openbanking.uk.forgerock.datamodel.common.FRAccountIdentifier;
+import com.forgerock.securebanking.common.openbanking.uk.forgerock.datamodel.common.FRRemittanceInformation;
+import com.forgerock.securebanking.common.openbanking.uk.forgerock.datamodel.vrp.FRWriteDomesticVrpDataInitiation;
+import com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.details.DomesticVrpPaymentConsentDetails;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.joda.time.Instant;
+import uk.org.openbanking.datamodel.common.OBActiveOrHistoricCurrencyAndAmount;
+import uk.org.openbanking.datamodel.vrp.OBDomesticVRPControlParameters;
+import uk.org.openbanking.datamodel.vrp.OBDomesticVRPControlParametersPeriodicLimits;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.details.ConsentDetailsConstants.intent.OAUTH2_CLIENT_NAME;
+import static com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.details.ConsentDetailsConstants.intent.OB_INTENT_OBJECT;
+import static com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.details.ConsentDetailsConstants.intent.members.*;
+import static com.forgerock.securebanking.openbanking.uk.rcs.converters.UtilConverter.isNotNull;
+
+/**
+ * Converter class to map {@link JsonObject} to {@link DomesticVrpPaymentConsentDetails}
+ */
+@Slf4j
+@NoArgsConstructor
+public class DomesticVrpPaymentConsentDetailsConverter {
+    private static volatile DomesticVrpPaymentConsentDetailsConverter instance;
+
+    /*
+     * Double check locking principle to ensure that only one instance 'DomesticVrpPaymentConsentDetailsConverter' is created
+     */
+    public static DomesticVrpPaymentConsentDetailsConverter getInstance() {
+        if (instance == null) {
+            synchronized (DomesticVrpPaymentConsentDetailsConverter.class) {
+                if (instance == null) {
+                    instance = new DomesticVrpPaymentConsentDetailsConverter();
+                }
+            }
+        }
+        return instance;
+    }
```

Comment:
It would be better to use spring to manage DI, when something needs a DomesticVrpPaymentConsentDetailsConverter then allow spring to provide it.

This is a much cleaner method and more flexible method. The code that uses the converter doesn't need to concern itself with obtaining one, avoiding static is also beneficial as test code can more easily supply a mock without having to resort to the messy mockStatic

Reply:
I have deleted the converters and moved the mapping of IDM object to ConsentDetails object expected by RCS-UI  implement the Factory Method Pattern and delegating the DI of components to spring.